axon/src/main/java/com/baeldung/axon/coreapi/queries/OrderUpdatesQuery.java

```diff
@@ -0,0 +1,13 @@
+package com.baeldung.axon.coreapi.queries;
+
+public class OrderUpdatesQuery {
+    private final String orderId;
+
+    public OrderUpdatesQuery(String orderId) {
+        this.orderId = orderId;
+    }
+
+    public String getOrderId() {
+        return orderId;
+    }
```

Comment:
Nit: I'd add an equals, hashcode and toString for completions sake.

Reply:
Maybe I could switch them to Records? I guess they should be fairly familiar and it means a lot less code. What do you think? Probably for consistency need to change anything that can be a record.

Reply:
That would actually be a lovely plan! 
I'd check with the Baeldung team on this, though.

Furthermore, I think it would warrant a change throughout all the immutable objects within the Axon Framework articles.
So that means rewriting more. A bit more work, but it does give consistency.

Reply:
I checked, since Java 17 is still little used in production, it's still better to avoid.